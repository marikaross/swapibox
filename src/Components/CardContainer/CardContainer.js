import React from 'react';
import PropTypes from 'prop-types';
import Card from '../Card/Card.js';

const CardContainer = ({cards}) => {
  console.log(cards)
  const allCards = cards.map(card => <Card key={card.name} {...card}/>);
  return (
    <div>
      {allCards}
    </div>

    )
}

export default CardContainer






CardContainer.propTypes = {
  cards: PropTypes.array.isRequired,
};