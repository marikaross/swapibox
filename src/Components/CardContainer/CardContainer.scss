.cardContainer {
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;
  width: 70%
  // background-color: black;
  justify-content: center;
}