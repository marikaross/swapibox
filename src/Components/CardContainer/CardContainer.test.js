import React from 'react';
import { shallow } from 'enzyme';
import CardContainer from './CardContainer.js';

describe('CardContainer', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const mockCards = 
    [{name: 'Bingo', 
      homeworld: 'farm', 
      species: 'canine', 
      population: 15 },
      {name: 'Wanda',
      homeworld: 'fish-bowl',
      species: 'fish',
      population: 1}
    ]

    const wrapper = shallow(<CardContainer cards={mockCards} />)
    expect(wrapper).toMatchSnapshot();

  })
})