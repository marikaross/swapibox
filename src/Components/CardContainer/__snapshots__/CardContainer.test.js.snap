// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot with all data passed in correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    cards={
        Array [
            Object {
              "homeworld": "farm",
              "name": "Bingo",
              "population": 15,
              "species": "canine",
            },
            Object {
              "homeworld": "fish-bowl",
              "name": "Wanda",
              "population": 1,
              "species": "fish",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          homeworld="farm"
          name="Bingo"
          population={15}
          species="canine"
/>,
        <Card
          homeworld="fish-bowl"
          name="Wanda"
          population={1}
          species="fish"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Bingo",
        "nodeType": "function",
        "props": Object {
          "homeworld": "farm",
          "name": "Bingo",
          "population": 15,
          "species": "canine",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Wanda",
        "nodeType": "function",
        "props": Object {
          "homeworld": "fish-bowl",
          "name": "Wanda",
          "population": 1,
          "species": "fish",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            homeworld="farm"
            name="Bingo"
            population={15}
            species="canine"
/>,
          <Card
            homeworld="fish-bowl"
            name="Wanda"
            population={1}
            species="fish"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Bingo",
          "nodeType": "function",
          "props": Object {
            "homeworld": "farm",
            "name": "Bingo",
            "population": 15,
            "species": "canine",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Wanda",
          "nodeType": "function",
          "props": Object {
            "homeworld": "fish-bowl",
            "name": "Wanda",
            "population": 1,
            "species": "fish",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
